File: app/src/main/java/net/onamap/android/dao/PhotoDao.java
package net.onamap.android.dao;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

public class PhotoDao {

    private List<Photo> photos;

    public PhotoDao(Context context) {
        photos = JsonParser.parse(context, "handstandsam");
    }

    public List<Photo> getPhotosForState(String stateName) {
        List<Photo> photosInState = new ArrayList<>();
        for (Photo photo : photos) {
            if (photo.country.equalsIgnoreCase("United States")) {
                if (photo.region != null && photo.region.equalsIgnoreCase(stateName)) {
                    photosInState.add(photo);
                }
            }
        }
        return photosInState;
    }
}

File: app/src/main/java/net/onamap/android/dao/PhotoDao.kt
package net.onamap.android.dao

import android.content.Context
import net.onamap.models.models.Photo

class PhotoDao {

    companion object {

        private lateinit var handstandSamPhotos: List<Photo>

        fun init(applicationContext: Context) {
            handstandSamPhotos = JsonParser.parse(applicationContext, "handstandsam")!!
            println(handstandSamPhotos)
        }

        fun getPhotosForState(stateName: String): List<Photo> {
            return getPhotosForState(handstandSamPhotos, stateName)
        }

        private fun getPhotosForState(allPhotos: List<Photo>, stateName: String?): List<Photo> {
            return allPhotos.filter {
                it.country == "United States" && it.region == stateName
            }
        }
    }
}